apps/integration/integration/ingests/crawler.py

```diff
@@ -0,0 +1,67 @@
+import docker
+from docker.models.containers import Container
+from opensearchpy import OpenSearch
+from integration.containers.preexisting import docker_compose
+from integration.ingests.index import IndexInfo
+import time
+
+PROFILE_TO_NAME_MAP = {"sort-one": "sycamore_crawler_http_sort_one", "sort-all": "sycamore_crawler_http_sort_all"}
+DEFAULT_INDEX_NAME = "demoindex0"
+
+
+class HttpCrawlerIndex:
+    def __init__(
+        self,
+        profile: str,
+        opensearch: OpenSearch,
+        importer: Container,
+    ):
+        self._profile = profile
+        self._opensearch = opensearch
+        self._importer = importer
+
+    def __enter__(self):
+        docker_client = docker.from_env()
+        service_name = self._get_service_name()
+        compose = docker_compose(services=[service_name])
+        files = set()
+        start_crawler_time = time.time()
+        compose.start()
+        crawler_container = compose.get_container(service_name=service_name, include_all=True)
+        crawler_container = docker_client.containers.get(crawler_container.ID)
+        assert crawler_container.wait().get("StatusCode") == 0, "Crawler container failed"
+        logs = [log.decode() for log in crawler_container.logs().splitlines()]
+        for log in reversed(logs):
+            if "Spider opened" in log:
+                break
+            if log.startswith("Store"):
+                pieces = log[6:].split(" as ")
+                file = pieces[1].strip()
+                if "unknown" not in file:
+                    files.add(file)
+        num_files = len(files)
+        importer_logs = self._importer.logs(stream=True, since=start_crawler_time)
+        for log in importer_logs:
+            log = log.decode()
+            if log.startswith("Successfully imported:"):
+                list_insides = log.rstrip()[len("Successfully imported: [") : -len("]")]
+                quoted_files = list_insides.split(",")
+                imported_files = [file.strip()[len("'/app/") : -len("'")] for file in quoted_files]
+                for file in imported_files:
+                    print(f"imported {file}")
+                    files.remove(file)
+                print(f"remaining files: {files}")
+            if len(files) == 0:
+                print("finished importing!")
+                break
+            if log.startswith("No changes") and len(files) != num_files:
+                raise RuntimeError(
+                    "Importer thinks there are no more files to ingest, but there are more files to ingest"
```

Comment:
Yeah, I'm pretty nervous about this whole deal. We could look at the crawler data volume I guess... that would also entail clearing it out before every run, which is probably worth doing anyway. Waiting for the ingestion to finish is harder, since the importer doesn't exit when it finishes. I guess we could maybe do some sort of opensearch query to get the set of filenames in the index, although I'm not sure that means all the chunks are in?

Reply:
looking into it further, it seems like there isn't a way to look at the data in the volume, at least on macos, because docker runs in a VM and gets like a drive partition that I can't access from outside (in my tests). We could do a bunch of `container.exec("ls")` calls, but this seems not much better. I'll also note that eric has plans to change how the importer does its detection type stuff so I think all of this code may have to get chucked regardless of whether I do something more robust with it. So I'm tempted to leave it as is.
You could also do something like repeatedly query opensearch until the index size stops changing, but I don't trust opensearch to update reliably enough for that to work